### PR TITLE
E06 t01 exploration page

### PR DESCRIPTION
**Pull request by Antoniatch**
# TASK TO EPIC PR
(Delete if no relevant)

## The feature / The problem
I can explore blogs and articles 

## What I've done / The solution
- Update discovery page with limits on the getAllBlogsAndArticles query
- Update blogs and articles pages with limits and offsets on the queries
- Made the pagination component dynamic
- Add scroll to top functionnality

## Scope of my changes
- App
- Pagination
- StaticPagination
- ListingBlogs
- Articles
- BlogT1 & BlogT2

## Task
- [e06-t01](link to issue)

## Type of change
- [x] New feature (feat)
- [ ] Fix issue (fix)
- [ ] Code refactorisation (refactor)
- [ ] Code testing (test)
- [ ] Add documentation (docs)

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my PR
- [x] I have removed not needed logs
- [x] I have remover TODO comments
- [ ] I have commented in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have been able to build my solution locally
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

# Communication
Pour l'instant j'ai mis la limite d'articles et de blogs à afficher sur leurs pages respectives à 6 (plus facile pour vérifier que la pagination fonctionne si on n'a pas 10 000 blogs et articles en bdd), mais elle sera à modifier ultérieurement

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -12,12 +12,16 @@ import Profil from './routes/profile'
...
-import { useContext } from 'react'
+import { useContext, useState } from 'react'
+import ScrollToTop from './utils/ScrollToTop'
 
 function App() {
   const { message } = useContext(NotificationContext)
+
   return (
     <BrowserRouter>
+      <ScrollToTop />
```
Comment (Holmes-EH) [inline src/App.tsx:23]: 🥳

File: src/main.tsx [hunk 2]
```diff
@@ -38,7 +38,9 @@ ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
     <ApolloProvider client={client}>
       <NotificationProvider>
         <UserProvider>
-          <App />
+          <>
```
Comment (Holmes-EH) [inline src/main.tsx:41]: Pourquoi le Fragment ?

File: src/routes/articles.tsx [hunk 3]
```diff
@@ -1,16 +1,76 @@
...
 const Articles = () => {
+  const { setMessage } = useContext(NotificationContext)
+
+  const [limit] = useState(6)
+  const [offset, setOffset] = useState(0)
+
+  const [numberOfPages, setNumberOfPages] = useState(1)
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const GET_ALL_ARTICLES_AND_TOTAL = gql`
```
Comment (Holmes-EH) [inline src/routes/articles.tsx:19]: Dans mes changes, et ceux de @AnasMama , il y a un dossier avec les queries utilisées dans l'app. On pourra probablement toutes les centraliser là-bas..
Plus tard quoi..

File: src/routes/blogs/ListingBlogs.tsx [hunk 4]
```diff
@@ -1,9 +1,72 @@
...
+const ListingBlogs = () => {
+  const { setMessage } = useContext(NotificationContext)
+
+  const [limit] = useState(6)
+  const [offset, setOffset] = useState(0)
+
+  const [numberOfPages, setNumberOfPages] = useState(1)
+  const [currentPage, setCurrentPage] = useState(1)
+
+  const GET_ALL_BLOGS_AND_TOTAL = gql`
```
Comment (Holmes-EH) [inline src/routes/blogs/ListingBlogs.tsx:18]: Same, on centralisera ?

**Review by Holmes-EH — APPROVED** [review]

File: src/main.tsx (hunk 2 above)
Comment (AnasMama) [reply src/main.tsx]: Meme question

**Review by AnasMama — APPROVED** [review]

**Review by Tbiarneix — APPROVED** [review]

**Pull request merged**